### PR TITLE
Adds support for s-ul

### PR DESCRIPTION
**Pull request by corin12355**
Implements native support for s-ul rather than having users use custom uploaders.

### CODE REVIEW

**Comment by Jaex** [comment]
Sorry but I can't accept uploader site which require registration and don't even have registration page. I designed custom uploader for that reason so people who wants to host their own uploader they can use it.

**Comment by corin12355** [comment]
Hi, this is basically something I did preemptively as we do have a registration page but we're just not finished doing frontend work and I guess I got ahead of myself https://n2.sys.s-ul.eu/. Would it be OK to resubmit the PR once this is live?

**Comment by Jaex** [comment]
I checked two different dictionary none of them knows what is "preemptively" means.

**Comment by corin12355** [comment]
http://dictionary.reference.com/browse/preemptive Definition 2, basically before we had the final product ready UI wise.

**Comment by Jaex** [comment]
I can't decide without see actual design because many people started to buy 5$ hosting and hosting their own free uploader clones so I must first see quality of uploader and then decide is it worth adding. Because I can't accept every random user made uploader which mainly closing few months later because of bandwidth issues.
But you can write here when your design is finished and I will reopen it if it is good for my standards :P

**Comment by corin12355** [comment]
I already have plans for things like bandwidth and scaling and don't plan on closing up shop any time soon but can do! Thanks :)

**Comment by corin12355** [comment]
Hey @Jaex we're mostly done and it's been pushed live now but we're wondering what you mean by design? We don't really plan to overcomplicate it by throwing CSS into the mix and we're quite happy with default bootstrap for now.

**Comment by Jaex** [comment]
I can accept now but because too long time passed merging ui won't be possible. Especially resx file because visual studio randomly orders things in it.

File: ShareX.UploadersLib/Forms/UploadersConfigForm.cs [hunk 1]
```diff
@@ -2498,6 +2512,12 @@ private void txtCustomUploaderLog_LinkClicked(object sender, LinkClickedEventArg
 
         #endregion Custom Uploaders
 
+        private void label1_Click(object sender, EventArgs e)
```
Comment (Jaex) [inline ShareX.UploadersLib/Forms/UploadersConfigForm.cs]: ?

Comment (corin12355) [inline ShareX.UploadersLib/Forms/UploadersConfigForm.cs]: Ah, my bad I didn't catch that. I believe it's just fluff left over from the form builder.

**Comment by corin12355** [comment]
Awesome, I'll pull the latest version and see if I can get things working again and able to be merged by the end of tonight. Thanks again!

As for the resx file would I just need to readd the UI elements again against the current version or is there something I'm doing majorly wrong? Apologies in advance if this is a stupid question, I'm a pretty bad C# developer haha

**Comment by Jaex** [comment]
You only had one textbox on ui so it will take few minutes for you to re add them instead of trying to merge.

**Comment by corin12355** [comment]
turns out I was a little too busy the past few days with work
I finished most of it again last night but general question, what way would be best? As I've never encountered this situation before. 
Rebasing and commiting? or just closing this pr, forking the repo again and adding my changes on top of it?

**Comment by Jaex** [comment]
I don't know git that much either, I mean I didn't do much pull requests.

**Comment by corin12355** [comment]
In that case, for ease I'll just recreate it, that way there's no issues. Thanks again!

**Pull request closed without merging**